blaze-client/src/main/scala/org/http4s/client/blaze/Http1Connection.scala

```diff
@@ -86,11 +91,22 @@ private final class Http1Connection(val requestKey: RequestKey,
   }
 
   @tailrec
-  def reset(): Unit = {
+  private def reset(): Unit = {
     stageState.get() match {
       case v@ (Running | Idle) =>
         if (stageState.compareAndSet(v, Idle)) parser.reset()
         else reset()
+      case t @ Tunneling(sslStage) =>
+        sendOutboundCommand(Command.Disconnect)
+        // TODO why doesn't this work?  The next request gets a bunch of ^U
```

Comment:
HELP!

Reply:
How did you see the bunch of ^U's? Was there a specific test?

I started to have a look at this by the way, but my knowledge of http4s is lacking at present. I've been trying to resolve the merge conflicts of this PR, mainly by using Cats instead of Scalaz. Going okay so far!

Reply:
Wow, sorry, just saw this comment.

It has been a while, but I think the ^U's happened on the second request to the same host.  I remember testing this manually in a hotel room, so I don't think there was a unit test yet.  I was just hacking at this point.

Reply:
No worries. I haven't had time to continue looking at this in the last week. A little stuck on converting Scalaz to Cats, but hopefully gaining understanding as I go. I might pop by the gitter room at some point in the coming week for some help with it :)